File: frontend/src/types/coins.ts
// backend/app/schemas/coins.py 1:1 대응 (docs/02-coding-conventions.md 9장)

export interface Coin {
  symbol: string;
  korean_name: string;
  english_name: string;
  current_price: string | null;
  change_rate: string | null;
}

export interface AvailableBalance {
  available_krw: string;
  available_quantity: string;
}

File: backend/app/schemas/coins.py
"""03-manual-trading 요청/응답 DTO — 코인 목록/가용잔고.

금액·비율 필드는 문자열로 반환한다 (02-coding-conventions.md 9절).
"""

from pydantic import BaseModel


class CoinResponse(BaseModel):
    symbol: str
    korean_name: str
    english_name: str
    current_price: str | None
    change_rate: str | None


class AvailableBalanceResponse(BaseModel):
    available_krw: str
    available_quantity: str
